runtime/common/BaseRemoteSimulatorQPU.h

```diff
@@ -11,15 +11,26 @@
 #include "common/ExecutionContext.h"
 #include "common/Logger.h"
 #include "common/RemoteKernelExecutor.h"
+#include "common/Resources.h"
 #include "common/RuntimeMLIR.h"
 #include "common/SerializedCodeExecutionContext.h"
 #include "cudaq.h"
+#include "cudaq/Optimizer/Builder/Runtime.h"
+#include "cudaq/Optimizer/Transforms/Passes.h"
 #include "cudaq/algorithms/gradient.h"
 #include "cudaq/algorithms/optimizer.h"
 #include "cudaq/platform/qpu.h"
 #include "cudaq/platform/quantum_platform.h"
+#include "mlir/ExecutionEngine/ExecutionEngine.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/Passes.h"
 #include <fstream>
 
+namespace nvqir {
+cudaq::Resources *getResourceCounts();
```

Comment:
we really can't be coupling all these libraries to the NVQIR library

Reply:
I can work around this here in the short term. For the longer term it may be nicer to attach the resource counter and/or relevant metadata to either the execution manager or platform to avoid needing the NVQIR library in the QPUs to nicely handle resource counting. Thoughts @1tnguyen?